lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -42,23 +84,41 @@ typedef _KeyDataResponseCallback = void Function(bool handled);
 class EnginePlatformDispatcher extends ui.PlatformDispatcher {
   /// Private constructor, since only dart:ui is supposed to create one of
   /// these.
-  EnginePlatformDispatcher._() {
+  EnginePlatformDispatcher() {
     _addBrightnessMediaQueryListener();
+    HighContrastSupport.instance.addListener(_updateHighContrast);
     _addFontSizeObserver();
+    registerHotRestartListener(dispose);
   }
 
   /// The [EnginePlatformDispatcher] singleton.
   static EnginePlatformDispatcher get instance => _instance;
-  static final EnginePlatformDispatcher _instance =
-      EnginePlatformDispatcher._();
+  static final EnginePlatformDispatcher _instance = EnginePlatformDispatcher();
 
   /// The current platform configuration.
   @override
   ui.PlatformConfiguration configuration = ui.PlatformConfiguration(
     locales: parseBrowserLanguages(),
     textScaleFactor: findBrowserTextScaleFactor(),
+    accessibilityFeatures: initializeAccessibilityFeatures(),
   );
 
+  /// Initialize accessibility features based on the current value of high contrast flag
+  static EngineAccessibilityFeatures initializeAccessibilityFeatures() {
+    final EngineAccessibilityFeaturesBuilder builder =
+        EngineAccessibilityFeaturesBuilder(0);
+    if (HighContrastSupport.instance.isHighContrastEnabled()) {
+      builder.highContrast = true;
+    }
+    return builder.build();
+  }
+
+  void dispose() {
+    _removeBrightnessMediaQueryListener();
+    _disconnectFontSizeObserver();
+    HighContrastSupport.instance.removeListener(_updateHighContrast);
```

Comment:
Oh, I see that this covers the hot-restart case. This is unusual because typically our singletons manage their own lifecycle w.r.t. to hot-restart, but in this case the lifecycle of `HighContrastSupport` is owned by `EnginePlatformDispatcher`. It not necessarily a bug, but could be confusing because it deviates from other similar classes. I think the initialization of `HighContrastSupport` may need to change for it to self-manage. Feel free to proceed how you see fit.

Reply:
Can you elaborate on "I think the initialization of HighContrastSupport may need to change for it to self-manage."?

Reply:
What I meant was that `HighContrastSupport` would need something like `ensureInitialized` method that creates the singleton and attaches a hot-restart listener that cleans up the singleton, similar to how `KeyboardBinding` does it.